### ISSUE TITLE
Possibility of rollback

### ISSUE THREAD

**Issue by MrReda** [mongobee/mongobee#69]
Right now we are in need of a process to rollback ,we have been using mongobee and its a good thing.
Why didn't you guys add this functionality ?
If so,i propose my self to implement it.

**Comment by inthegarage** [comment]
If you do then this is probably your best option:
https://docs.mongodb.com/manual/tutorial/configure-a-delayed-replica-set-member/

**Comment by MrReda** [comment]
inthegarage  thank you for your feedback,actually i was talking about adding the possibility of rollbacking the script in case of error happening during the deployment just like the Liquibase functionality,so thus we will be able to get back to previous working schema.
Let me know if this is not understandable for you.

**Comment by inthegarage** [comment]
@MrReda I don't think you can roll-back scripts in Mongo, that functionality is non-existent. So for instance if you delete something, then it's gone for good.
From what I've read the replica set is the way of doing this. So you write to a replica set and then when you are happy that becomes live. I'm not sure there's any other way.

**Comment by MrReda** [comment]
@inthegarage We will add a script to rollback in case of error just like liquibase.
I don't know if you are aware of that.Something similar to that could be very helpful.

**Comment by inthegarage** [comment]
@MrReda  I don't know how that could work. If I'm writing changeLogs how can mongobee possibly know what to rollback? Or are you saying we would just right a simple script to reverse any changes? If we are doing a wildcard delete/update it would not be possible.

**Comment by MrReda** [comment]
We will just put a script to rollback the latest played scripts.

**Comment by MrReda** [comment]
It depends about the use case.We could right a specific script in case of
rollback.
But let me explain to you one thing,when we talk about script rollback we
mean e.g.:adding new collection and later if something happen ,the
collection will be romoved (rollbacked),the same is true with users
creation or records as well.
We could make a poc and see if it worth it.

**Comment by NearteamABE** [comment]
I think it could be a good idea to propose a mechanism to revert a ChangeSet so that it is possible to go back to a lower software version. If we compare with Liquibase,  when you use a raw SQL script (which is comparable here to mongobee since we have full access to the mongo api) it is up to the developer responsability to provide the rollback script.
